aipl/interpreter.py

```diff
@@ -287,10 +291,26 @@ def prep_output(aipl,
         if isinstance(out, Table):
             return out
         else:
-            ret = Table(parent=in_row._table)
-            ret.rows = [{'__parent': in_row, varname:v} for v in out]
-            for k in ret.rows[0].keys():  # assumes first row has same keys as every other row
-                ret.add_column(Column(k))
+            if in_row is None:
+                parent_table = None
+                parent_row = None
+            elif isinstance(in_row, Table):
+                parent_table = None
+                parent_row = None
+            elif isinstance(in_row, LazyRow):
+                parent_table = in_row._table
+                parent_row = in_row
+            else:
+                raise Exception(f'unknown type for in_row: {type(in_row)}')
+
+            rows = []
+            for v in out:
+                d = {'__parent': parent_row} if parent_row is not None else {}
+                d[varname] = v
+                rows.append(d)
+            ret = Table(rows, parent=parent_table)
+#            for k in ret.rows[0].keys():  # assumes first row has same keys as every other row
+#                ret.add_column(Column(k))
```

Comment:
@saulpw do you remember why you needed this diff block?

Reply:
The `in_row` was None for the arity=0, rankout=1.5 case for `!python-input`.